### ISSUE TITLE
gramex.data.filter does not cache sql

### ISSUE THREAD

**Issue by pratapvardhan** [gramener/gramex#179]
Internally, in `data.py` we are using
```
data = gramex.cache.query(text(query), engine, state, params=all_params)

# elsewhere in cache.py (text(query), engine) is used for cache lookup

def query(sql, engine, state=None, **kwargs):
    key = (sql, engine.url)
    current_status = _cache.get(key, {}).get('status', None)
```

Now, `text(query)` returns a new object for same `query` text. Hence, internal caching of this key doesn't work.

Probably use a cache lookup for `text(query)` or internally in `cache.py` use sql text for key instead?

**Comment by sanand0** [comment]
@pratapvardhan Let's discuss when we speak next, please?

**Comment by pratapvardhan** [comment]
fixed via df97890
